src/opnsense/mvc/app/views/OPNsense/IPsec/key_pairs.volt

```diff
@@ -0,0 +1,123 @@
+   notice, this list of conditions and the following disclaimer in the
+   documentation and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+#}
+
+<script>
+  $(document).ready(function () {
+    const $applyLegacyConfig = $('#applyLegacyConfig');
+    const $applyLegacyConfigProgress = $('#applyLegacyConfigProgress');
+    const $responseMsg = $('#responseMsg');
+    const $dirtySubsystemMsg = $('#dirtySubsystemMsg');
+
+    // Helper method to fetch the current status of the legacy subsystem for viewing/hiding the "pending changes" alert
+    function updateLegacyStatus() {
+      ajaxCall('/api/ipsec/legacy-subsystem/status', {}, function (data, status) {
+        if (data['isDirty']) {
+          $responseMsg.addClass('hidden');
+          $dirtySubsystemMsg.removeClass('hidden');
+        } else {
+          $dirtySubsystemMsg.addClass('hidden');
+        }
+      });
+    }
+
+    // Apply config in legacy subsystem
+    $applyLegacyConfig.on('click', function (e) {
+      e.preventDefault();
+
+      $applyLegacyConfig.prop('disabled', true);
+      $applyLegacyConfigProgress.addClass('fa fa-spinner fa-pulse');
+
+      ajaxCall('/api/ipsec/legacy-subsystem/applyConfig', {}, function (data, status) {
+        // Preliminarily hide the "pending changes" alert and display the response message
+        $dirtySubsystemMsg.addClass('hidden');
+        $responseMsg.removeClass('hidden').text(data['message']);
+
+        // Reset the state of the "apply changes" button
+        $applyLegacyConfig.prop('disabled', false);
+        $applyLegacyConfigProgress.removeClass('fa fa-spinner fa-pulse');
+
+        // Fetch the current legacy subsystem status to ensure changes have been processed
+        updateLegacyStatus();
+      });
+    });
+
+    // Initialize grid for displaying and manipulating key pairs
+    const $grid = $('#grid-key-pairs').UIBootgrid({
+      search: '/api/ipsec/key-pairs/searchItem',
+      get: '/api/ipsec/key-pairs/getItem/',
+      set: '/api/ipsec/key-pairs/setItem/',
+      add: '/api/ipsec/key-pairs/addItem/',
+      del: '/api/ipsec/key-pairs/delItem/',
+    });
+
+    // Refresh status of legacy subsystem when grid has changed
+    $grid.on('loaded.rs.jquery.bootgrid', function() {
```

Comment:
please don't click apply - you need to add that manually if you want to (due to missing multiline support):
```suggestion
    $grid.on('loaded.rs.jquery.bootgrid', updateLegacyStatus);
```

Reply:
Sure, makes sense - updated.